streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -243,19 +248,19 @@ public void punctuate(ProcessorNode node, long timestamp) {
         log.trace("{} Punctuating processor {} with timestamp {}", logPrefix, node.name(), timestamp);
 
         try {
-            node.processor().punctuate(timestamp);
+            node.punctuate(timestamp);
         } catch (KafkaException ke) {
             throw new StreamsException(String.format("Exception caught in punctuate. taskId=%s processor=%s", id,  node.name()), ke);
         } finally {
             processorContext.setCurrentNode(null);
         }
     }
 
-
     /**
      * Commit the current task state
      */
     public void commit() {
+        long startNs = time.nanoseconds();
```

Comment:
We could save this `nanoseconds` with `maybeRecord` pattern as well with DEBUG level.